pkg/build/util/generate.go

```diff
@@ -118,8 +118,8 @@ func SubstituteImageReferences(build *buildapi.Build, oldImage string, newImage
 		build.Parameters.Strategy.DockerStrategy.BaseImage = newImage
 	case build.Parameters.Strategy.Type == buildapi.STIBuildStrategyType &&
 		build.Parameters.Strategy.STIStrategy != nil &&
-		build.Parameters.Strategy.STIStrategy.Image == oldImage:
-		build.Parameters.Strategy.STIStrategy.Image = newImage
+		build.Parameters.Strategy.STIStrategy.From.Name == oldImage:
```

Comment:
i don't think this is how this should be handled.  It needs some more thought, but "oldImage" is not the same type as "From.Name" so comparing and assigning them directly is not valid.

also it's possible that Image is specified, so you need to handle that case.